recipes-bsp/imx-secure-enclave/imx-secure-enclave_git.bb

```diff
@@ -1,31 +1,59 @@
-SUMMARY = "NXP i.MX Secure Enclave Userspace Library"
-DESCRIPTION = "NXP i.MX Secure Enclave Userspace Library"
-SECTION = "base"
+# Copyright (C) 2021-2024 NXP
 
+SUMMARY = "NXP i.MX SECURE ENCLAVE library"
+DESCRIPTION = "NXP IMX SECURE ENCLAVE library"
+SECTION = "base"
 LICENSE = "BSD-3-Clause"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=8636bd68fc00cc6a3809b7b58b45f982"
 
-DEPENDS = "openssl"
+DEPENDS = "mbedtls openssl"
+
+SRC_URI = "git://github.com/NXP/imx-secure-enclave.git;protocol=https;branch=${SRCBRANCH}"
+SRCBRANCH = "lf-6.12.3_1.0.0"
+SRCREV = "1130c8bb820881ad037ba3f060e7fa70635fae3c"
 
 inherit systemd
 
-SRC_URI = "git://github.com/NXP/imx-secure-enclave.git;protocol=https;branch=lf-6.6.52_2.2.0"
-SRCREV = "dffbb844e86f4a49058ffbb40548474059969c27"
+S = "${WORKDIR}/git"
 
-PV = "lf-6.6.52_2.2.0"
+PLAT ?= "UNDEFINED"
 
-S = "${WORKDIR}/git"
+PLAT:mx8ulp-nxp-bsp = "ele"
+PLAT:mx91-nxp-bsp = "ele"
+PLAT:mx93-nxp-bsp = "ele"
 
-EXTRA_OEMAKE += "PLAT=ele"
+PLAT:mx8x-nxp-bsp = "seco"
+PLAT:mx95-nxp-bsp = "seco"
 
 TARGET_CC_ARCH += "${LDFLAGS}"
 
-do_install () {
-    oe_runmake DESTDIR=${D} install
-    rm -rf ${D}/usr/share
-}
+EXTRA_OEMAKE = " \
+    PLAT=${PLAT} \
+    OPENSSL_PATH=${STAGING_INCDIR} \
+    MBEDTLS_PATH=${STAGING_DIR_HOST}${datadir}/mbedtls-source \
+"
+EXTRA_OEMAKE:append:mx8x-nxp-bsp = " \
+    COMPATIBLE_MACHINE=mx8dxl-nxp-bsp \
+"
 
 SYSTEMD_AUTO_ENABLE = "disable"
 SYSTEMD_SERVICE:${PN} = "nvm_daemon.service"
+SYSTEMD_SERVICE:${PN}:mx95-nxp-bsp = ""
+
+do_install() {
+	oe_runmake -C ${S} DESTDIR=${D} install_tests
+}
+
+do_install:append:mx95-nxp-bsp() {
+    # Remove common content that is to be installed by imx-secure-enclave
```

Comment:
This doesn't make sense since the recipes are now combined. Combining the recipes suggests that you can just drop the append.

Reply:
We can certainly put both together. However, we will need to have a variable to control whether we need to remove or not those files.

Reply:
With two recipes, the source install duplicated headers, a README, and a script, which caused a build break due to installing the same file multiple times to the rootfs. With one combined recipe, this specific problem doesn't exist.